crates/ruff_python_formatter/src/expression/string.rs

```diff
@@ -543,10 +543,17 @@ fn preferred_quotes(
                             // `""` or `''`
                             chars.next();
 
-                            if chars.peek().copied() == Some(configured_quote_char) {
-                                // `"""` or `'''`
-                                chars.next();
-                                uses_triple_quotes = true;
+                            match chars.peek().copied() {
+                                Some(c) if c == configured_quote_char => {
+                                    // `"""` or `'''`
+                                    chars.next();
+                                    uses_triple_quotes = true;
+                                }
+                                Some(_) => {}
+                                None => {
+                                    // Handle `''' ""'''`
+                                    uses_triple_quotes = true;
+                                }
```

Comment:
Can you explain the fix? 

I'm asking because I would expect `preferred_quotes` to return single here because it contains two `"`... Changing the quotes to `"` would require two escapes... which is more than when keeping the single quotes.